src/openforms/registrations/contrib/json_dump/plugin.py

```diff
@@ -95,34 +96,107 @@ def process_variables(submission: Submission, values: JSONObject):
-                case False, int():  # pragma: no cover
-                    raise ValueError(
-                        f"Cannot have multiple attachments ({n_attachments}) for a "
-                        "single file component."
-                    )
+            component = get_component(variable, submission)
+            assert component is not None
+
+            match component["type"]:
+                case "file":
+                    encoded_attachments = [
+                        {
+                            "file_name": attachment.original_name,
+                            "content": encode_attachment(attachment),
+                        }
+                        for attachment in submission.attachments
+                        if attachment.form_key == key
+                    ]
+
+                    base_schema = {
+                        "title": component["label"],
+                        "type": "object",
+                        "properties": {
+                            "file_name": {"type": "string"},
+                            "content": {"type": "string", "format": "base64"},
+                        },
+                        "required": (
+                            ["file_name", "content"]
+                            if len(encoded_attachments) != 0
+                            else []
+                            # No required properties when there are no attachments
+                        ),
+                        "additionalProperties": False,
+                    }
+
+                    match (
+                        component.get("multiple", False),
+                        n_attachments := len(encoded_attachments),
+                    ):
+                        case False, 0:
+                            value = None
+
+                            base_schema = {"title": component["label"], "type": "null"}
+                        case False, 1:
+                            value = encoded_attachments[0]
+                        case True, int():
+                            value = encoded_attachments
+
+                            base_schema = to_multiple(base_schema)
+                        case False, int():  # pragma: no cover
+                            raise ValueError(
+                                f"Cannot have multiple attachments ({n_attachments}) "
+                                "for a single file component."
+                            )
+                    values[key] = value
+                    schema["properties"][key] = base_schema
+                case "radio":
+                    data_src = component.get("openForms", {}).get("dataSrc")
+                    if data_src != "variable":
+                        # Only components where another variable is used as a data
+                        # source need to be processed, so skip this one
+                        continue
+
+                    choices = [options["value"] for options in component["values"]]
```

Comment:
In these instances I think it's okay to `cast(...)` to the more specific type hint since we already pattern matched it to the right type